RecoLocalCalo/EcalRecProducers/plugins/alpaka/AmplitudeComputationCommonKernels.h

```diff
@@ -0,0 +1,503 @@
+#ifndef RecoLocalCalo_EcalRecProducers_plugins_alpaka_AmplitudeComputationCommonKernels_h
+#define RecoLocalCalo_EcalRecProducers_plugins_alpaka_AmplitudeComputationCommonKernels_h
+
+#include <cstdlib>
+#include <limits>
+#include <alpaka/alpaka.hpp>
+
+#include "CondFormats/EcalObjects/interface/alpaka/EcalMultifitConditionsDevice.h"
+#include "DataFormats/EcalDigi/interface/alpaka/EcalDigiDeviceCollection.h"
+#include "DataFormats/EcalRecHit/interface/alpaka/EcalUncalibratedRecHitDeviceCollection.h"
+#include "CondFormats/EcalObjects/interface/EcalPulseShapes.h"
+#include "DataFormats/EcalDigi/interface/EcalDataFrame.h"
+#include "DataFormats/EcalDigi/interface/EcalMGPASample.h"
+#include "DataFormats/EcalRecHit/interface/EcalUncalibratedRecHit.h"
+#include "FWCore/Utilities/interface/CMSUnrollLoop.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/traits.h"
+
+#include "DeclsForKernels.h"
+#include "../EigenMatrixTypes_gpu.h"
```

Comment:
I think that "by policy" this should go under `RecoLocalCalo/EcalRecProducers/interface/`.
And it also adds a trivial dependency on `CUDADataFormats/EcalRecHitSoA`.

I understand that in this case it is safe because it only defines `constexpr` constants and `typedef`s.

Still, would it be feasible to rearrange the files ?

Reply:
`EigenMatrixTypes_gpu.h` is also used by the CUDA version. I can move it to `RecoLocalCalo/EcalRecProducers/interface/` but the dependency would still be there.
One could think about moving `CUDADataFormats/EcalRecHitSoA/interface/RecoTypes.h` to `DataFormats/EcalRecHit/interface` to get rid of that.

Reply:
Would it work to move `CUDADataFormats/EcalRecHitSoA/interface/RecoTypes.h` to `RecoLocalCalo/EcalRecProducers/interface/` ?

Reply:
> `RecoTypes.h` is also used in other headers in `CUDADataFormats/EcalRecHitSoA`. So that package would gain a dependency on `RecoLocalCalo/EcalRecProducers`. Is that OK?

Adding such dependence on `CUDADataFormats` would not be good. Moving the `RecoTypes.h` to `DataFormats` sounds a better option.

Reply:
Moved `RecoTypes.h` to `DataFormats/EcalRecHit`.